lib/sequenceserver.rb

```diff
@@ -373,6 +388,26 @@ def tuple?(data)
       end
     end
 
+    get '/get_report/' do
+      format = params[:format]
+      specifiers = params[:specifiers]
+
+      rfile  = File.join(Dir.tmpdir, params[:report])
+
+      file_name = "sequenceserver_report_#{Time.now.strftime('%H%M')}.#{format}"
+      file = Tempfile.new file_name
+
+      case format
+      when 'xml'
+        file.write(BLAST.blast_formatter(rfile, OUTFMT['xml']))
```

Comment:
Use `Hash#fetch` instead of `[]`. Rescue `KeyError`, and re-raise it as 400 Bad Request.